nixos/modules/services/monitoring/glances.nix

```diff
@@ -0,0 +1,79 @@
+{
+  pkgs,
+  config,
+  lib,
+  ...
+}:
+let
+  cfg = config.services.glances;
+
+  inherit (lib)
+    escapeShellArgs
+    maintainers
+    mkEnableOption
+    mkOption
+    mkIf
+    mkPackageOption
+    ;
+
+  inherit (lib.types)
+    bool
+    listOf
+    port
+    str
+    ;
+in
+{
+  options.services.glances = {
+    enable = mkEnableOption "Glances";
+
+    package = mkPackageOption pkgs "glances" { };
+
+    port = mkOption {
+      description = "Port the server will isten on.";
+      type = port;
+      default = 61208;
+    };
+
+    openFirewall = mkOption {
+      description = "Open port in the firewall for glances.";
+      type = bool;
+      default = false;
+    };
+
+    commandlineArgs = mkOption {
+      type = listOf str;
+      default = [ "--webserver" ];
+      example = [
+        "--webserver"
+        "--disable-webui"
+      ];
+      description = ''
+        Set commandline arguments to pass to glances. See
+        https://glances.readthedocs.io/en/latest/cmds.html.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    environment.systemPackages = [ cfg.package ];
+
+    systemd.services."glances" = {
+      description = "Glances";
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+
+      serviceConfig = {
+        Type = "simple";
+        User = "root";
```

Comment:
must we run this as root?
I'd personally really appreciate seeing the user/group be specific to the service and be configurable via a user and a group option.

Reply:
I guess that should be possible (and I like it that way too :)) but I guess that would also require to add some specific capabilities to be able to monitor certain things, I'll see what I can find in the glances code/wiki/doc

Reply:
User and group added, been running it since yesterday and seems to work fine together with the hardening as well